source/yats/include/yats/task_container.h

```diff
@@ -80,8 +91,15 @@ class task_container : public abstract_task_container
 
     void run() override
     {
-        m_options->make_updates_visible(&m_task);
-        invoke(std::make_index_sequence<helper::input_count>());
+        try
+        {
+            m_options->make_updates_visible(&m_task);
+            invoke(std::make_index_sequence<helper::input_count>());
+        } catch (const std::exception&)
+        {
+            m_error = std::current_exception();
```

Comment:
This might not be thread safe.

Reply:
a task does not need to be thread safe, does it? this is executed on exactly one thread

Reply:
I think you are right. There seems to be an instance of std::current_execption for each individual thread.